Jenkinsfile

```diff
@@ -118,6 +130,35 @@ pipeline {
                 }
             }
         }
+        stage('Publish to Test-pypi') {
+            when {
+                branch 'main'
+            }
+            steps {
+                script {
+                    sh "twine check dist/csvcubed*.whl"
+
+                    withCredentials([usernamePassword(credentialsId: 'testpypi-robons', usernameVariable:'TWINE_USERNAME', passwordVariable: 'TWINE_PASSWORD')]) {
+                        sh 'twine upload -r testpypi dist/csvcubed*.whl'
+                    }
+                }
+            }
+        }
+        stage('Publish to Pypi') {
+            when {
+                buildingTag()
+                tag pattern: "v\\d+\\.\\d+\\.\\d+(-RC\\d)?", comparator: "REGEXP"
+            }
+            steps {
+                script {
+                    sh "twine check dist/csvcubed*.whl"
+
+                    withCredentials([usernamePassword(credentialsId: 'pypi-robons', usernameVariable:'TWINE_USERNAME', passwordVariable: 'TWINE_PASSWORD')]) {
+                        sh 'twine upload dist/csvcubed*.whl'
+                    }
```

Comment:
@robons This is self-explanatory from the name of usernameVariable and passwordVariable. But I would like to double-check. The TWINE_USERNAME and TWINE_PASSWORD are env variables pointing to the actual username and password securely stored in jenkins right? And they are not the actual password and username?